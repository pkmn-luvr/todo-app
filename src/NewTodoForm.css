.NewTodoForm {
    display: flex;
    flex-direction: column;
    gap: 5px;
  }
  
  .NewTodoForm input,
  .NewTodoForm button {
    padding: 10px;
    border-radius: 5px;
    border: 2px solid #a7bbc7;
  }
  
  .NewTodoForm button {
    background-color: #d6e6f2;
    color: #333;
  }
  