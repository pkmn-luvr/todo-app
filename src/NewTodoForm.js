import React, { useState } from 'react';
import './NewTodoForm.css';

function NewTodoForm({ addTodo }) {
  const [task, setTask] = useState('');

  const handleSubmit = (e) => {
    e.preventDefault();
    addTodo({ task, id: Date.now() });
    setTask('');
  };

  return (
    <form onSubmit={handleSubmit}>
  <label htmlFor="new-todo">Add a new task:</label>
  <input
    id="new-todo"
    type="text"
    value={task}
    onChange={(e) => setTask(e.target.value)}
    placeholder="Add a new task"
  />
  <button type="submit">Add Todo</button>
</form>
  );
}

export default NewTodoForm;
