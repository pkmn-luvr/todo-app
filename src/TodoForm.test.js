import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import NewTodoForm from './NewTodoForm';

describe('NewTodoForm Component Tests', () => {
  
  // Smoke
    test('renders without crashing', () => {
    render(<NewTodoForm />);
  });

  // Snapshot
  test('matches snapshot', () => {
    const { asFragment } = render(<NewTodoForm />);
    expect(asFragment()).toMatchSnapshot();
  });

  // Tests for Business Logic - Does submitting the form update DOM? 
  test('form submission calls addTodo prop', () => {
    const addTodoMock = jest.fn();
    const { getByLabelText, getByText } = render(<NewTodoForm addTodo={addTodoMock} />);
    fireEvent.change(getByLabelText('Add a new task:'), { target: { value: 'New Task' } });
    fireEvent.click(getByText('Add Todo'));
    expect(addTodoMock).toHaveBeenCalledWith({ task: 'New Task', id: expect.any(Number) });
  });
});
