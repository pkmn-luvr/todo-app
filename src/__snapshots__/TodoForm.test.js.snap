// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewTodoForm Component Tests matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="new-todo"
    >
      Add a new task:
    </label>
    <input
      id="new-todo"
      placeholder="Add a new task"
      type="text"
      value=""
    />
    <button
      type="submit"
    >
      Add Todo
    </button>
  </form>
</DocumentFragment>
`;
